src/modules/core/gateway.service.ts

```diff
@@ -197,31 +200,42 @@ export class GatewayService {
   //Plus potentially applying a score to the encoder..
   async failJob(job_id, node_id) {
     const jobInfo = await this.jobs.findOne({
-      id: job_id
+      id: job_id,
     })
-    if(!jobInfo) {
-      return;
+    if (!jobInfo) {
+      return
     }
-    if(jobInfo.status === JobStatus.ASSIGNED || jobInfo.status === JobStatus.RUNNING) { 
-      if(jobInfo.assigned_to === node_id) {
+    if (jobInfo.status === JobStatus.ASSIGNED || jobInfo.status === JobStatus.RUNNING) {
+      if (jobInfo.assigned_to === node_id) {
         await this.activity.changeState({
           job_id: job_id,
           new_status: JobStatus.QUEUED,
           assigned_to: null,
           meta: {
-            reason: 'failed'
-          }
-        })
-        await this.jobs.findOneAndUpdate({
-          id: job_id
-        }, {
-          $set: {
-            status: JobStatus.QUEUED,
-            assigned_date: null,
-            assigned_to: null,
-            last_pinged: null
-          }
+            reason: 'failed',
+          },
         })
+        await this.jobs.findOneAndUpdate(
+          { id: job_id },
+          {
+            $set: {
+              status: JobStatus.QUEUED,
+              assigned_date: null,
+              assigned_to: null,
+              last_pinged: null,
+            },
+            $inc: { num_fails: 1 },
+          },
+        )
+        // read the same job count for failures
+        const jobInfo = await this.jobs.findOne(
+          { id: job_id, status: JobStatus.QUEUED },
+          { sort: { created_at: 1 } },
```

Comment:
Why is sort needed here?

Reply:
It's a copy-pasted code from above lines which were written earlier to get the record of the job.